client/src/app/dashboard/datatable-notification/datatable-notification.component.html

```diff
@@ -0,0 +1,40 @@
+<div class="mat-elevation-z8">
+  <table class="notifTable" mat-table #table [dataSource]="dataSource" matSort aria-label="Elements">
+    <!-- type Column -->
+    <ng-container matColumnDef="type">
+      <th mat-header-cell *matHeaderCellDef mat-sort-header>Type</th>
+      <td mat-cell *matCellDef="let row"><mat-icon>{{row.type}}</mat-icon></td>
+    </ng-container>
+
+    <!-- Name notif Column -->
+    <ng-container matColumnDef="notifName">
+      <th mat-header-cell *matHeaderCellDef mat-sort-header>Notification</th>
+      <td mat-cell *matCellDef="let row">{{row.notifName}}</td>
+    </ng-container>
+
+    <!-- Date name Column -->
+    <ng-container matColumnDef="dateTime">
+      <th mat-header-cell *matHeaderCellDef mat-sort-header>Date and Time</th>
+      <td mat-cell *matCellDef="let row">{{row.dateTime}}</td>
+    </ng-container>
+
+
+    <!-- Details Column -->
+    <ng-container matColumnDef="details">
+      <th mat-header-cell *matHeaderCellDef mat-sort-header>Details</th>
+      <td mat-cell *matCellDef="let row">
+        <button mat-raised-button (click)="openDialog()">details</button>
```

Comment:
The openDialog method is missing in the datatable-notification.component.ts . I saw that you injected a MatDialog component in your component, I believe you just forgot to write the openDialog() method opening this dialog? :)

Reply:
Ok I've added the openDialog function but I will implement it later.